src/ripple/app/main/Application.cpp

```diff
@@ -1034,6 +1035,54 @@ class ApplicationImp
                     << "Remaining free disk space is less than 512MB";
                 signalStop ();
             }
+
+            DatabaseCon::Setup dbSetup = setup_DatabaseCon(*config_);
+            boost::filesystem::path dbPath = dbSetup.dataDir / TxnDBName;
+            boost::system::error_code ec;
+            std::uint64_t dbSize = boost::filesystem::file_size(dbPath, ec);
+            if (ec)
+            {
+                JLOG(m_journal.error())
+                    << "Error checking transaction db file size: "
+                    << ec.message();
+                return;
+            }
+
+            auto db = mTxnDB->checkoutDb();
+            static auto const pageSize = [&]{
```

Comment:
could these lambdas be made into a single `[&] get_pragma_value(char const* fieldname) -> uint32_t` and then called three times (`page_size`, `max_page_count`, `page_count`)?